File: plugin/src/org/jetbrains/yesod/julius/JuliusFileType.java
package org.jetbrains.yesod.julius;

/**
 * @author Leyla H
 */
import com.intellij.openapi.fileTypes.LanguageFileType;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.haskell.icons.HaskellIcons;

import javax.swing.*;

public class JuliusFileType extends LanguageFileType {
    public static final JuliusFileType INSTANCE = new JuliusFileType();
    @NonNls
    public static final String DEFAULT_EXTENSION = "julius";

    private Icon myIcon;

    private JuliusFileType() {
        super(JuliusLanguage.INSTANCE);
        myIcon = HaskellIcons.HAMLET;
    }

    @NotNull
    public String getName() {
        return "Julius file";
    }

    @NotNull
    public String getDescription() {
        return "Julius file";
    }

    @NotNull
    public String getDefaultExtension() {
        return DEFAULT_EXTENSION;
    }

    public Icon getIcon() {
        return myIcon;
    }

    public String getCharset(@NotNull VirtualFile file, final byte[] content) {
        return "UTF-8";
    }
}

File: plugin/src/org/jetbrains/yesod/julius/JuliusFileType.kt
package org.jetbrains.yesod.julius

/**
 * @author Leyla H
 */
import com.intellij.openapi.fileTypes.LanguageFileType
import com.intellij.openapi.vfs.VirtualFile
import org.jetbrains.annotations.NonNls
import org.jetbrains.haskell.icons.HaskellIcons

import javax.swing.*

public class JuliusFileType private constructor() : LanguageFileType(JuliusLanguage.INSTANCE) {

    private val myIcon: Icon

    init {
        myIcon = HaskellIcons.HAMLET
    }

    override fun getName(): String {
        return "Julius file"
    }

    override fun getDescription(): String {
        return "Julius file"
    }

    override fun getDefaultExtension(): String {
        return DEFAULT_EXTENSION
    }

    override fun getIcon(): Icon? {
        return myIcon
    }

    override fun getCharset(file: VirtualFile, content: ByteArray): String? {
        return "UTF-8"
    }

    companion object {
        public val INSTANCE: JuliusFileType = JuliusFileType()

        public val DEFAULT_EXTENSION: String = "julius"
    }
}